app/Tenant.php

```diff
@@ -3,63 +3,68 @@
     {
-        $this->customer = $customer;
-        $this->website = $website ?? $customer->websites->first();
-        $this->hostname = $hostname ?? $customer->hostnames->first();
+        $this->website = $website;
+        $this->hostname = $hostname;
         $this->admin = $admin;
     }
 
-    public function delete()
+    public static function delete($name)
     {
-        app(HostnameRepository::class)->delete($this->hostname, true);
-        app(WebsiteRepository::class)->delete($this->website, true);
-        app(CustomerRepository::class)->delete($this->customer, true);
+        $baseUrl = env('APP_URL_BASE');
+        $name = "{$name}.{$baseUrl}";
+        if ($tenant = Hostname::where('fqdn', $name)->firstOrFail()) {
+            app(HostnameRepository::class)->delete($tenant, true);
+            app(WebsiteRepository::class)->delete($tenant->website, true);
+            return "Tenant {$name} successfully deleted.";
+        }
     }
 
-    public static function createFrom($name, $email, $password = null): Tenant
+    public static function deleteByFqdn($fqdn)
     {
-        // create a customer
-        $customer = new Customer;
-        $customer->name = $name;
-        $customer->email = $email;
+        if ($tenant = Hostname::where('fqdn', $fqdn)->firstOrFail()) {
+            app(HostnameRepository::class)->delete($tenant, true);
+            app(WebsiteRepository::class)->delete($tenant->website, true);
+            return "Tenant {$fqdn} successfully deleted.";
+        }
+    }
 
-        app(CustomerRepository::class)->create($customer);
+    public static function registerTenant($name, $email, $password): Tenant
+    {
+        // Convert all to lowercase
+        $name = strtolower($name);
+        $email = strtolower($email);
 
-        // associate the customer with a website
         $website = new Website;
-        $website->customer()->associate($customer);
         app(WebsiteRepository::class)->create($website);
 
         // associate the website with a hostname
         $hostname = new Hostname;
-        $baseUrl = config('app.url_base');
+        $baseUrl = env('APP_URL_BASE');
         $hostname->fqdn = "{$name}.{$baseUrl}";
-        $hostname->customer()->associate($customer);
         app(HostnameRepository::class)->attach($hostname, $website);
+
         // make hostname current
-        app(Environment::class)->hostname($hostname);
+        app(Environment::class)->tenant($hostname->website);
 
-        $admin = static::makeAdmin($name, $email, $password ?: str_random());
+        // Make the registered user the default Admin of the site.
+        $admin = static::makeAdmin($name, $email, $password);
 
-        return new Tenant($customer, $website, $hostname, $admin);
+        return new Tenant($website, $hostname, $admin);
     }
 
     private static function makeAdmin($name, $email, $password): User
```

Comment:
Not sure what you intend doing here but i see potential issues aas no user is made the Admin. Code should be 

private static function makeAdmin($name, $email, $password): User
    {
        $admin = User::create(['name' => $name, 'email' => $email, 'password' => Hash::make($password)]);
        $admin->guard_name = 'web';
        $admin->assignRole('admin');

        return $admin;
    }

Reply:
Thanks for taking a look. Can't quite see the change you are suggesting? The createTenant command  creates an admin user with the email address passed in and names it as the hostname.

To be honest I prefer your implementation with a proper registration front-end but the tutorial does make it clear that has been left out to keep things simple. I've tried to do enough to make the tutorial work but not require a massive update. I see a lot of queries from people stuck with the change to 5.2 and @ashokgelal asked for a PR so that he could update the tutorial. I've tried to do it with the least change possible so he will have the least change needed to his tutorial. (he may well decide to extend it of course!)

It took a while for me to find what you'd done and it helped a lot. Hopefully this will get a few other people off the starting blocks and through this thread they can pick up with the work you are doing to move on after the tutorial :)

Reply:
You left the function empty

private static function makeAdmin($name, $email, $password): User {
return $admin;
}

Idea is to have someone as the Admin on the tenant database and make sure someone can log in to help setup others unless you have a different implementation in mind. Else no one will be able to log into the new tenant created.

Reply:
Please ignore. Mistake was from me.

Reply:
No worries - I thought the function was there!